queries/powershell/highlights.scm

```diff
@@ -0,0 +1,324 @@
+; Punctuation
+;------------
+[
+  "("
+  ")"
+  "["
+  "]"
+  "{"
+  "}"
+] @punctuation.bracket
+
+(hash_literal_expression
+  [
+    "@{"
+    "}"
+  ] @punctuation.special)
+
+(array_expression
+  [
+    "@("
+    ")"
+  ] @punctuation.special)
+
+(member_access
+  "." @punctuation.special)
+
+(pipeline
+  "|" @punctuation.special)
+
+[
+  ","
+  ";"
+  (empty_statement)
+  (command_argument_sep)
+] @punctuation.delimiter
+
+; Keywords
+;---------
+[
+  "if"
+  "elseif"
+  "else"
+  "switch"
+] @keyword.conditional
+
+[
+  "foreach"
+  "for"
+  "while"
+  "do"
+  "until"
+  "in"
+  "break"
+  "continue"
+] @keyword.repeat
+
+[
+  "function"
+  "filter"
+  "workflow"
+  "exit"
+  "trap"
+  "param"
+  "inlinescript"
+] @keyword.function
+
+"return" @keyword.return
+
+[
+  "class"
+  "enum"
+] @keyword.type
+
+(class_attribute) @keyword.modifier
+
+[
+  "throw"
+  "try"
+  "catch"
+  "finally"
+] @keyword.exception
+
+[
+  "parallel"
+  "sequence"
+] @keyword.coroutine
+
+[
+  "param"
+  "dynamicparam"
+  "begin"
+  "process"
+  "end"
+  ; TODO: not supported by parser yet, can be used to declare constants
+  "data"
+] @keyword
+
+; Operators
+;----------
+[
+  "-and"
+  "-or"
+  "-xor"
+  "-band"
+  "-bor"
+  "-bxor"
+  "+"
+  "-"
+  "/"
+  "\\"
+  "%"
+  "*"
+  ".."
+  "-not"
+  "-bnot"
+  (pre_increment_expression)
+  (pre_decrement_expression)
+  (post_increment_expression)
+  (post_decrement_expression)
+  (comparison_operator)
+  (assignement_operator)
+] @operator
+
+; Generic types
+;--------------
+(string_literal) @string
+
+(integer_literal) @number
+
+(real_literal) @number.float
+
+(variable) @variable
+
+((variable) @variable.builtin
+  (#any-of? @variable.builtin
+    "$$" "$?" "$^" "$_" "$args" "$ConsoleFileName" "$EnabledExperimentalFeatures" "$Error" "$Event"
+    "$EventArgs" "$EventSubscriber" "$ExecutionContext" "$false" "$foreach" "$HOME" "$Host" "$input"
+    "$IsCoreCLR" "$IsLinux" "$IsMacOS" "$IsWindows" "$LASTEXITCODE" "$Matches" "$MyInvocation"
+    "$NestedPromptLevel" "$null" "$PID" "$PROFILE" "$PSBoundParameters" "$PSCmdlet" "$PSCommandPath"
+    "$PSCulture" "$PSDebugContext" "$PSEdition" "$PSHOME" "$PSItem" "$PSScriptRoot" "$PSSenderInfo"
+    "$PSUICulture" "$PSVersionTable" "$PWD" "$Sender" "ShellId" "$StackTrace" "$switch" "$this"
+    "$true"))
+
+((variable) @variable.builtin
+  (#lua-match? @variable.builtin "^\$env:"))
+
+(comment
+  (#set! priority 90)) @comment @spell
```

Comment:
Is this really necessary? We prefer to rely on pattern ordering instead.

Reply:
I think no, my bad, I think  I misunderstood [Markup] section: `(use priority 90 for blocks with injections)`
[Markup]: https://github.com/nvim-treesitter/nvim-treesitter/blob/master/CONTRIBUTING.md#markup

will remove it

Reply:
Yes, _if_ that is needed (e.g., if you inject C into a verbatim block; otherwise you'll have the `@markup.raw` highlight "bleeding through").